### PR TITLE
insert example using-routes-todo-app

### PR DESCRIPTION
**Pull request by davidsonsns**
based on the Point-Cards example. I used localStorage to store the state

issue #50

### CODE REVIEW

File: examples/using-routes-todo-app/index.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+app.addActions({todos: TodoActions})
+app.addRoute('/', home)
+app.addRoute('/todo', todo)
+app.addRoute('/results', results)
+
+app.engine.addListener((store) => {
+  localStorage.setItem('todos', JSON.stringify(store.todos))
+})
```
Comment (JRJurman) [inline examples/using-routes-todo-app/index.js]: Nice use of `addListener`!

That being said, we don't need this if the links do `window.history.pushState({}, '', href)`
It means that page-refreshes will clear all the information, but for right now, I think I'd rather do that then use the engine's addListener function.

Comment (JRJurman) [reply examples/using-routes-todo-app/index.js]: Making this change would also require all the `<a>` tags to be rewritten as tags with a `onclick` that does the `pushState`

File: examples/using-routes-todo-app/index.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  const onCompleteTodo = (index) => () => {
+    actions.completeTodo(index)
+  }
+
+  return html`
+    <div>
+      <div class="center">
+        <h1> Tram-One Todos 🚋 </h1>
+        <a href="/">HOME</a> | <a href="/results">RESULTS</a>
+      </div>
```
Comment (JRJurman) [inline examples/using-routes-todo-app/index.js]: to reduce how many times we write this, lets make a custom element which is the header with all the links in it (even if it a bit redundant)

**Comment by davidsonsns** [comment]
I'm glad to hear that you liked it!

your review was really good, I'll do it.

I did not know of the existence of `pushState` in `history`, thanks for the explanation

File: examples/using-routes-todo-app/index.js (hunk 1 above)
Comment (davidsonsns) [reply examples/using-routes-todo-app/index.js]: I did not understand what you meant. so I should remove `addListener` and use `pushState`?

File: examples/using-routes-todo-app/elements/HeaderPage.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+module.exports = ({title, subtitle = undefined}, children) => {
+    const contentSubtitle = subtitle ? html`<h2> ${subtitle} </h2>` : null
+  return html`
+    <div>
+      <h1> ${title} 🚋 </h1>
+      ${contentSubtitle}
+    </div>
+  `
+}
```
Comment (JRJurman) [inline examples/using-routes-todo-app/elements/HeaderPage.js]: oh ho! I like this 👍

File: examples/using-routes-todo-app/index.js (hunk 1 above)
Comment (JRJurman) [reply examples/using-routes-todo-app/index.js]: I'm actually not 100% sold that we should change to this, so I'm approving the PR for now. We may change it in the future, but for now, this example is really good, so I'd rather get it in, and clean it up later if-need-be.

Comment (davidsonsns) [reply examples/using-routes-todo-app/index.js]: @JRJurman  I think if we leave both options would be interesting. maybe this same example implementing `pushState`

I really enjoyed the experience, I will continue to contribute with certainty

**Pull request merged**